src/renderer/display_list.rs

```diff
@@ -72,18 +72,22 @@ impl<'a> fmt::Debug for DisplayList<'a> {
 
 impl<'a> Display for DisplayList<'a> {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-        let max_lineno = self.body.iter().fold(0, |max, set| {
+        let max_lineno = self.body.iter().fold(None, |max, set| {
             set.display_lines.iter().fold(max, |max, line| match line {
-                DisplayLine::Source { lineno, .. } => cmp::max(lineno.unwrap_or(0), max),
+                DisplayLine::Source { lineno, .. } => match (max, lineno) {
+                    (Some(max), Some(lineno)) => Some(cmp::max(max, *lineno)),
+                    (Some(max), None) => Some(max),
+                    (None, Some(lineno)) => Some(*lineno),
+                    (None, None) => None,
```

Comment:
You can do `std::cmp::max(max, lineno)` without a match.  `Some` compares higher than `None`